apache/zeppelin
# [ZEPPELIN-1293] Re-create Livy session if it's lost

### What is this PR for?
This patch restarts Livy session when it's lost.

### What type of PR is it?
Improvement/Bug Fix

### Todos

### What is the Jira issue?
https://issues.apache.org/jira/browse/ZEPPELIN-1293

### How should this be tested?
 * Configure `livy.server.session.timeout` parameter in Livy
 * Start new Livy session through Zeppelin
 * Wait for Livy session to expire
 * See that new Livy session is started when running something in Zeppelin

### Questions:
* Does the licenses files need update?  - No
* Is there breaking changes for older versions?  - No
* Does this needs documentation?  - No
